src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2235,20 +2240,101 @@ private void VisitBinaryOperatorChildren(BoundBinaryOperator node)
         protected virtual void VisitBinaryOperatorChildren(ArrayBuilder<BoundBinaryOperator> stack)
         {
             var binary = stack.Pop();
-            VisitRvalue(binary.Left);
 
-            while (true)
+            // Only the leftmost operator of a left-associative binary operator chain can learn from a conditional access on the left
+            // For simplicity, we just special case it here.
+            if (VisitPossibleConditionalAccess(binary.Left, out var stateWhenNotNull)
+                && canLearnFromOperator()
+                && (isNullableValueTypeConversion(binary.Right) || binary.Right.ConstantValue is object))
```

Comment:
>isNullableValueTypeConversion(binary.Right) || binary.Right.ConstantValue is object) [](start = 20, length = 84)

nit: consider extracting this (and below) to a `isKnownNotNull(binary.Right)` or similar helper

Reply:
Done